dataland-frontend/src/components/resources/frameworkDataSearch/frameworkPanel/MultiLayerDataTableFrameworkPanel.vue

```diff
@@ -192,11 +192,23 @@ async function loadDataForDisplay(
   singleDataMetaInfoToDisplay?: DataMetaInformation
 ): Promise<DataAndMetaInformation<FrameworkDataType>[]> {
   const dataControllerApi = getFrameworkDataApiForIdentifier(props.frameworkIdentifier, apiClientProvider) as
-    | BaseFrameworkDataApi<FrameworkDataType>
+    | PublicFrameworkDataApi<FrameworkDataType>
     | undefined;
   if (dataControllerApi) {
```

Comment:
```suggestion
  if (!dataControllerApi) throw new Error(`No data controller found for framework ${props.frameworkIdentifier}`)
  if (!singleDataMetaInfoToDisplay) return (await dataControllerApi.getAllCompanyData(assertDefined(companyId))).data;
  let singleDataset;
  ...
```
vermeidet die Verschachtelungen